--- a/Lib/prompt_toolkit/application/application.py
+++ b/Lib/prompt_toolkit/application/application.py
@@ -1604,6 +1604,12 @@
     else:
         # GraalPy has the functions, but they don't work
         have_ctypes_signal = sys.implementation.name != "graalpy"
+        if have_ctypes_signal:
+            try:
+                pythonapi.PyOS_getsig
+                pythonapi.PyOS_setsig
+            except AttributeError:
+                have_ctypes_signal = False
 
     if have_ctypes_signal:
         # PyOS_sighandler_t PyOS_getsig(int i)
